designcompose/src/androidTest/kotlin/com/android/designcompose/JniNoNetworkTests.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose
+
+import androidx.test.platform.app.InstrumentationRegistry
+import java.io.IOException
+import java.net.ConnectException
+import java.net.SocketException
+import kotlin.test.assertFailsWith
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.anyOf
+import org.hamcrest.core.IsInstanceOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class JniNoNetworkTests {
+    @Before
+    fun setup() {
+        InstrumentationRegistry.getInstrumentation()
+            .uiAutomation
+            .executeShellCommand("svc wifi disable")
+        InstrumentationRegistry.getInstrumentation()
+            .uiAutomation
+            .executeShellCommand("svc data disable")
+    }
+    @Test
+    fun networkFailure() {
+        val exception =
+            assertFailsWith<IOException> {
+                LiveUpdateJni.jniFetchDoc("DummyDocId", dummyFigmaTokenJson)
+            }
```

Comment:
Do you need to check for an auth token in setup() I tried running this test and got the following:

java.lang.AssertionError: Expected an exception of class java.io.IOException to be thrown, but was com.android.designcompose.AccessDeniedException: Invalid Authentication Token
at kotlin.test.AssertionsKt__AssertionsImplKt.AssertionErrorWithCause(AssertionsImpl.kt:117)

Reply:
Hm, no, this one is specifically meant to test a network connection error to Figma. The setup should turn off networking on your emulator, and the cleanup should turn it back on. Could you try it again?